### ISSUE TITLE
Tooltip - Design Doc

### ISSUE THREAD

**Issue by mcarrano** [patternfly/patternfly-org#1108]
Introduce design documentation for Tooltip.

Related issue: #113

**Comment by jgiardino** [comment]
@matthewcarleton, @rachael-phillips and I talked about Tooltips in the a11y review meeting today. 

The predominant pattern for accessible tooltips is to use `aria-describedby` or `aria-labelledby` to associate the tooltip text with the element that has focus, but there are some gaps where this doesn't work, which are noted in https://github.com/patternfly/patternfly-react/issues/789. These gaps are mainly cases where tooltips are used with static elements.

The conclusion that we came to is that instead of trying to create a solution that addresses these accessibility gaps, we should instead define in the design documentation that using tooltips on static elements is not recommended, and include suggestions for alternate solutions.

I've listed below some ideas on what we could suggest instead. @matthewcarleton also mentioned some examples where his team recently revised similar designs to use an alternate solution, so it would be great to see what his team has come up with, too.

Alternate suggestions for tooltips on static elements:

- Display the tooltip contents inline next to the element. For example, an icon that represents status could have the status stated (specifically for status that is not obvious from the icon).
- Display the contents in a popover
- For recurring tooltips, especially when the same contents could be repeated in a tooltip more than 10 times, consider creating a legend (i.e. a single place where this information can be accessed). Creating additional tab stops for static elements creates an additional burden for keyboard-only users.
  - Note that this case often occurs where tooltips are used to describe icons, and that an accessible name for these icons is still needed for screen readers. The recommended method for these is to include screen-reader-only text adjacent to the icon element.

**Comment by abigaeljamie** [comment]
If we decide to create a doc on this, can we please include these UX writing guidelines for tooltips?  

- Keep your tooltips clear and concise. Use the fewest number of words you can without sacrificing meaning. 
- If the tooltip is a full sentence, end it with a period.   
- If information is needed for a user to complete a task (like a password character requirement), don’t hide it in a tooltip. Display it on the page instead.

**Comment by mcarrano** [comment]
@mmenestr Do you think it makes sense to just fold this into the embedded assistance document that you are currently working on? It already references tooltips.

**Comment by mmenestr** [comment]
Yes, that definitely makes sense! Just added @abigaeljamie's writing guidelines in the doc.
Do you think we should also add anything about accessibility for tooltips in there or no?
